sympy/polys/polyroots.py

```diff
@@ -1012,6 +1012,10 @@ def _try_heuristics(f):
                             if not result:
                                 for root in _try_decompose(f):
                                     _update_dict(result, root, 1)
+                        else:
+                            for factor, k in factors:
```

Comment:
It seems to me that there is only one factor (`f`) and `k == 1` at this point. The outer loop is probably not needed.

Reply:
You're right. I have removed the outer loop in my branch.

Reply:
It looks like you haven't pushed your branch here to refresh this. I still see the outer loop.